src/app/CacheManagers/CounterTableDisplay.tsx

```diff
@@ -128,21 +107,116 @@ const CounterTableDisplay = (props: { setCountersCount: (number) => void; isVisi
     });
   };
 
+  const updateRows = (counters: Counter[]) => {
+    let rows: Array<any> | React.SetStateAction<any>;
+
+    if (counters.length == 0) {
+      rows = [];
+      setActions([]);
+    } else {
+      rows = counters;
+      setActions(
+        selectedFilters.counterType === CounterType.STRONG_COUNTER ? strongCountersActions : weakCountersActions
+      );
+    }
+    setRows(rows);
+  };
+
+  const onSelectFilter = (event, selection) => {
```

Comment:
@karesti I've used a lot of if-else for this function, let me if there is any refracting possible for this.